rust/ql/lib/codeql/rust/frameworks/stdlib/Stdlib.qll

```diff
@@ -14,7 +14,7 @@ private import codeql.rust.internal.PathResolution
  */
 private class StartswithCall extends Path::SafeAccessCheck::Range, CfgNodes::MethodCallExprCfgNode {
   StartswithCall() {
-    this.getAstNode().(Resolvable).getResolvedPath() = "<crate::path::Path>::starts_with"
+    this.getMethodCallExpr().getStaticTarget().getCanonicalPath() = "<std::path::Path>::starts_with"
```

Comment:
See https://github.com/github/codeql/pull/20226 .  This was untested, often doesn't work due to type inference gaps, but has proven to be essentially correct.